### PR TITLE
SLING-11736 Add "ensure nodes" as replacement of "create path" with

### PR DESCRIPTION
**Pull request by kwin**
stricter semantics of adjusting types of existing nodes

### CODE REVIEW

File: src/main/java/org/apache/sling/repoinit/parser/operations/CreateNode.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                }
+                sb.append(")");
+            }
+        }
+
+        // FIXME: see SLING-10238 for type and quoted values that cannot be generated
+        //        exactly as they were originally defined in repo-init
+        try (Formatter formatter = new Formatter()) {
+            if (lines.isEmpty()) {
+                formatter.format("create path %s%s%n",  defaultTypeStr, sb.toString());
```
Comment (kwin) [inline src/main/java/org/apache/sling/repoinit/parser/operations/CreateNode.java:73]: this is a copy & paste mistake

**Comment by kwin** [comment]
I will merge tomorrow if there are no concerns.

File: src/main/java/org/apache/sling/repoinit/parser/operations/CreatePath.java [hunk 2]
```diff
@@ -25,7 +25,12 @@
 import org.jetbrains.annotations.NotNull;
 import org.osgi.annotation.versioning.ProviderType;
 
+/**
+ * @deprecate Use {@link EnsureNodes} instead.
```
Comment (bdelacretaz) [inline src/main/java/org/apache/sling/repoinit/parser/operations/CreatePath.java:29]: should be @deprecated, right?

File: src/main/java/org/apache/sling/repoinit/parser/operations/EnsureNodes.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    }
+    
+    @Override
+    protected String getParametersDescription() {
+        return pathDef.toString();
+    }
+
+    @NotNull
+    @Override
+    public String asRepoInitString() {
```
Comment (bdelacretaz) [inline src/main/java/org/apache/sling/repoinit/parser/operations/EnsureNodes.java:51]: I think this code is (mostly) duplicated from CreatePath, shouldn't the common code be factored out? The test coverage is quite good, refactoring doesn't looks risky to me.

File: src/test/java/org/apache/sling/repoinit/parser/test/OperationToStringVisitor.java [hunk 4]
```diff
@@ -223,6 +224,16 @@ public void visitCreatePath(CreatePath cp) {
         }
     }
 
+    @Override
+    public void visitCreateNode(EnsureNodes cn) {
```
Comment (bdelacretaz) [inline src/test/java/org/apache/sling/repoinit/parser/test/OperationToStringVisitor.java:228]: As the operation class is called EnsureNodes I think this method should be called `visitEnsureNodes` for consistency

**Review by bdelacretaz — COMMENTED** [review]
I've left a few comments, nothing major but I think it's worth addressing them

**Pull request merged**